### PR TITLE
chore: using sfdx-core utill function

### PR DESCRIPTION
**Pull request by vamsimundra**
### What does this PR do?
It uses checkLightningDomain of sfdx-core in force:org:open command.
### What issues does this PR fix or reference?
@W-9369342@

### CODE REVIEW

File: src/commands/force/org/open.ts [hunk 1]
```diff
@@ -51,7 +50,17 @@ export class OrgOpenCommand extends SfdxCommand {
...
     }
     // we actually need to open the org
-    await this.checkLightningDomain(url);
+    try {
+      this.ux.startSpinner(messages.getMessage('domainWaiting'));
+      await checkLightningDomain(url);
+    } catch (err) {
+      const domain = `https://${/https?:\/\/([^.]*)/.exec(url)[1]}.lightning.force.com`;
+      const timeout = new Duration(new Env().getNumber('SFDX_DOMAIN_RETRY', 240), Duration.Unit.SECONDS);
+      this.logger.debug(`Did not find IP for ${domain} after ${timeout.seconds} seconds`);
```
Comment (vamsimundra) [inline src/commands/force/org/open.ts:59]: What do you think is the best way to throw the error?
Irrespective of what error we are catching, I am throwing  "domainTimeoutError" as it was used previously. What do you think of this?

Comment (maggiben) [reply src/commands/force/org/open.ts:59]: I think the error is quite descriptive, the function will throw after the timeout+retries end so is safe to say it fits the description, the key `domainTimeoutError` seems descriptive enough to me.

File: src/commands/force/org/open.ts [hunk 2]
```diff
@@ -5,11 +5,10 @@
  * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 import { EOL } from 'os';
-import { URL } from 'url';
 
 import { flags, FlagsConfig, SfdxCommand } from '@salesforce/command';
-import { Messages, Org, MyDomainResolver, SfdxError, sfdc } from '@salesforce/core';
-import { Env, Duration } from '@salesforce/kit';
+import { Messages, Org, checkLightningDomain, SfdxError } from '@salesforce/core';
```
Comment (maggiben) [inline src/commands/force/org/open.ts:10]: Be aware that I forgot to add an export for `checkLightningDomain` here: https://github.com/forcedotcom/sfdx-core/blob/main/src/exported.ts and thus Im not sure you'll be able to load this dependency this way maybe if you do: `import checkLightningDomain from '@salesforce/core/util/checkLightningDomain'` maybe.

**Review by maggiben — APPROVED** [review]

File: test/commands/force/org/open.test.ts [hunk 3]
```diff
@@ -7,6 +7,7 @@
 import { $$, expect, test } from '@salesforce/command/lib/test';
 import { Org, MyDomainResolver, Messages } from '@salesforce/core';
 import { stubMethod } from '@salesforce/ts-sinon';
+// import { SfdxError } from '../../../../../sfdx-core/lib/sfdxError';
```
Comment (WillieRuemmele) [inline test/commands/force/org/open.test.ts:10]: let's remove this comment

File: src/commands/force/org/open.ts (hunk 2 above)
Comment (mshanemc) [reply src/commands/force/org/open.ts:10]: we're keeping the checkLightningDomain for backward-compatibility reasons. This command can use the checkLightningDomain on sfdcUrl class (validates url, etc).

https://github.com/forcedotcom/sfdx-core/pull/420/files

File: src/commands/force/org/open.ts [hunk 4]
```diff
@@ -51,7 +50,21 @@ export class OrgOpenCommand extends SfdxCommand {
       return output;
     }
     // we actually need to open the org
-    await this.checkLightningDomain(url);
+    try {
+      this.ux.startSpinner(messages.getMessage('domainWaiting'));
+      await checkLightningDomain(url);
+    } catch (err) {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access,  @typescript-eslint/no-unsafe-call
+      if (err.message && err.message.includes('timeout')) {
```
Comment (mshanemc) [inline src/commands/force/org/open.ts:58]:
```suggestion
      if (err.message?.includes('timeout')) {
```
evaluates to undefined (falsy to the `if`) when there is no message.

File: src/commands/force/org/open.ts [hunk 5]
```diff
@@ -51,7 +50,21 @@ export class OrgOpenCommand extends SfdxCommand {
...
+        const domain = `https://${/https?:\/\/([^.]*)/.exec(url)[1]}.lightning.force.com`;
+        const timeout = new Duration(new Env().getNumber('SFDX_DOMAIN_RETRY', 240), Duration.Unit.SECONDS);
```
Comment (mshanemc) [inline src/commands/force/org/open.ts:60]: where did the 240 come from?  Is that the default but it's in core?

**Review by mshanemc — APPROVED** [review]

**Review by mshanemc — CHANGES_REQUESTED** [review]
needs a bump of core so the tests pass.

**Review by mshanemc — APPROVED** [review]

**Pull request merged**